app/Icepeak.hs

```diff
@@ -33,7 +38,15 @@ installHandlers core serverThread =
 
 main :: IO ()
 main = do
-  core <- Core.newCore
+  config <- execParser configInfo
+  maybeValue <- withFile (cDataFile config) ReadMode BS.hGetContents
+  -- load the persistent data from disk
+  -- maybeValue <- BS.readFile "icepeak.json"
+
+  let value = case eitherDecode maybeValue of
+                Left _msg  -> Object mempty
```

Comment:
Instead of silently ignoring the error, it would be better to fail loudly and refuse to start the program. If it uses a default value instead, and there is a write, the previous file (which failed to load) will be overwritten, which might result in data loss. If the file fails to load for some reason, the user might want to back up the corrupt file, or try and repair it manually. For some users deleting the data and starting from an empty value automatically would be the right thing to do, but that would be up to them to decide and automate. (E.g. a systemd `OnFailure` hook that removes the file and restarts.)

Reply:
Agreed, I now error out for both missing as well as invalid files.